CMakeLists.txt

```diff
@@ -1,24 +1,45 @@
 cmake_minimum_required(VERSION 3.1)
 set(CMAKE_CXX_STANDARD 14)
 
-project(client)
+project(client LANGUAGES CXX C VERSION 2.7.0)
+include(FeatureSummary)
+find_package(ECM 5.50.0  NO_MODULE)
+set_package_properties(ECM PROPERTIES TYPE REQUIRED DESCRIPTION "Extra CMake Modules." URL "https://projects.kde.org/projects/kdesupport/extra-cmake-modules")
+feature_summary(WHAT REQUIRED_PACKAGES_NOT_FOUND FATAL_ON_MISSING_REQUIRED_PACKAGES)
+
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules ${ECM_MODULE_PATH})
```

Comment:
Prepend is rather new and this is how it was done always since the beginning of time :D

Reply:
... and manually patching buildsystems was also needed since the beginning of time. 
I prefer `list(APPEND ...)` but if you insist on `set(...)` please add the initial value there.